File: HackerRank/30 Days of Code/Day20_Sorting.cs
// Day 20: Sorting
// https://www.hackerrank.com/challenges/30-sorting/problem

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {

    static void Main(String[] args) {
        int n = Convert.ToInt32(Console.ReadLine());
        string[] a_temp = Console.ReadLine().Split(' ');
        int[] a = Array.ConvertAll(a_temp,Int32.Parse);
        int swaped = 0;
        for (int i = 0; i < n; i++){
            for (int j = 0; j < n - 1; j++){
                if (a[j] > a[j+1]){
                    int temp = a[j+1];
                    a[j+1] = a[j];
                    a[j] = temp;
                    swaped++;
                }
            }
        }
        Console.WriteLine("Array is sorted in {0} swaps.", swaped);
        Console.WriteLine("First Element: {0}", a[0]);
        Console.WriteLine("Last Element: {0}", a[n-1]);
    }
}

// 버블 정렬 구현하고 교체 횟수 구하기

File: HackerRank/30 Days of Code/Day20_Sorting.py
# Day 20: Sorting
# https://www.hackerrank.com/challenges/30-sorting/problem

import sys

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))

swaped = 0
for i in range(n):
    for j in range(n - 1):
        if a[j] > a[j+1]:
            a[j], a[j+1] = a[j+1], a[j]
            swaped += 1

print("Array is sorted in", swaped, "swaps.")
print("First Element:", a[0])
print("Last Element:", a[n-1])


# 버블 정렬 구현하고 교체 횟수 구하기
